mirdata/msd.py

```diff
@@ -0,0 +1,373 @@
+# -*- coding: utf-8 -*-
+"""Million Song Dataset Loader
+
+The Million Song Dataset is a freely-available collection of audio features and metadata for a million contemporary popular music tracks.
+
+More details can be found on http://millionsongdataset.com/
+
+Attributes:
+    DATASET_DIR (str): The directory name for Million Song Dataset. Set to `'msd'`.
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tables
+
+import mirdata.utils as utils
+import mirdata.download_utils as download_utils
+
+DATASET_DIR = "MSD"
+
+
+DATA = utils.LargeData("msd_index.json")
+
+
+class Track(object):
+    """MSD Track class
+
+    Args:
+        track_id (str): Track id of the track
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Attributes:
+        Please refer to http://millionsongdataset.com/
+        """
+
+    def __init__(self, track_id, data_home=None):
+        self.track_id = track_id
+        self.data_filename = _filename(track_id, data_home=data_home)
+
+        if not os.path.exists(self.data_filename):
+            raise ValueError("Invalid track ID: %s" % track_id)
+
+        with tables.open_file(self.data_filename, mode="r") as f:
+            self.analysis = MSDAnalysis(f.root.analysis)
+            self.metadata = MSDMetadata(f.root.metadata)
+            self.musicbrainz = MSDMusicBrainz(f.root.musicbrainz)
+
+        self._data_home = data_home
+
+    def __repr__(self):
+        return (
+            "msd.Track("
+            + "analysis={analysis}, "
+            + "metadata={metadata}, "
+            + "musicbrainz={musicbrainz})"
+        ).format(
+            analysis=self.analysis, metadata=self.metadata, musicbrainz=self.musicbrainz
+        )
+
+
+class MSDAnalysis(object):
```

Comment:
I know it's a pain but can you add documentation for each of these classes?

Reply:
Yep, will do!